File: python/src/tests/t_next_slave.py
#!/usr/bin/python3
#
# Id:
#
# We test a bit of the atrshmlog here.
#
# This is for the first starter, so only the basic things.

import sys
import atrshmlog

target = int(sys.argv[1])

result = atrshmlog.attach()

atrshmlog.sleep_nanos(1000000000)

akcount = 0

slave = atrshmlog.get_next_slave_local(0)

while slave != 0:
    akcount = akcount + 1
    slave = atrshmlog.get_next_slave_local(slave)

print('next slave : ' + str(akcount) + ' : ')

s = 0

for i in range (0, 8) :
    s = atrshmlog.create_slave()


print('next slave : ' + str(s) + ' : ')

atrshmlog.sleep_nanos(1000000000)

akcount = 0

slave = atrshmlog.get_next_slave_local(0)

while  slave != 0 :
    akcount = akcount + 1
    print('next slave : ' + str(akcount) + ' : ')
    print('next slave : ' + str(slave) + ' : ')
    
    tid = atrshmlog.get_slave_tid(slave)
    
    print('next slave : ' + str(tid) + ' : ')
    
    if akcount == target :
        targetslave = slave
        
    slave = atrshmlog.get_next_slave_local(slave)


print('next slave : ' + str(akcount) + ' : ')

if targetslave != 0 :
    atrshmlog.turn_slave_off(targetslave)

    atrshmlog.sleep_nanos(1000000000)

akcount = 0

slave = atrshmlog.get_next_slave_local(0)

while  slave != 0 :
    akcount = akcount + 1
    print('next slave : ' + str(akcount) + ' : ')
    print('next slave : ' + str(slave) + ' : ')
    
    tid = atrshmlog.get_slave_tid(slave)
    
    print('next slave : ' + str(tid) + ' : ')
    
    slave = atrshmlog.get_next_slave_local(slave)



print (' ')

exit(0)

# end of test

File: perl/src/tests/t_next_slave.pl
#!/usr/bin/perl
#
# $Id:$
#
# perl test driver

use Atrshmlog;

$target = shift @ARGV;

$result = Atrshmlog::attach();

Atrshmlog::sleep_nanos(1000000000);

$akcount = 0;

$slave = undef;

$slave = Atrshmlog::get_next_slave_local($slave);

while ($slave) {
    ++ $akcount;
    $slave = Atrshmlog::get_next_slave_local($slave);
}

print "next slave : $akcount : \n";

$s = 0;

for ($i = 0; $i < 8 ; ++$i) {
    $s = Atrshmlog::create_slave();
}

print "next slave : $s : \n";

Atrshmlog::sleep_nanos(1000000000);

$targetslave = undef;

$akcount = 0;

$slave = undef;

$slave = Atrshmlog::get_next_slave_local($slave);

while ($slave) {
    ++ $akcount;

    print "next slave : $akcount : \n";

    print "next slave : $$slave : \n";

    $tid = Atrshmlog::get_slave_tid($slave);
    
    print "next slave : $tid : \n";

    if ($akcount == $target) {
	$targetslave = $slave;
    }
    
    $slave = Atrshmlog::get_next_slave_local($slave);
}

print "next slave : $akcount : \n";

if ($targetslave) {
    Atrshmlog::turn_slave_off($targetslave);

    Atrshmlog::sleep_nanos(1000000000);

    $akcount = 0;

    $slave = undef;

    $slave = Atrshmlog::get_next_slave_local($slave);

    while ($slave) {
	++ $akcount;

	print "next slave : $akcount : \n";

	print "next slave : $$slave : \n";

	$tid = Atrshmlog::get_slave_tid($slave);
    
	print "next slave : $tid : \n";

	$slave = Atrshmlog::get_next_slave_local($slave);
    }
    
}     
     
print " \n";

exit (0);

# end of main
